### PR TITLE
Minimum Price Support in UI + Minor contract fix

### PR DESCRIPTION
**Pull request by bhgames**

### CODE REVIEW

File: js/packages/common/src/actions/auction.ts [hunk 1]
```diff
@@ -306,6 +306,104 @@ export class WinnerLimit {
...
+  22: number;
+  23: number;
+  24: number;
+  25: number;
+  26: number;
+  27: number;
+  28: number;
+  29: number;
+  30: number;
+  31: number;
```
Comment (bhgames) [inline js/packages/common/src/actions/auction.ts:344]: Forgive me @bartosz-lipinski

File: js/packages/web/src/components/AuctionNumbers/index.tsx [hunk 2]
```diff
@@ -98,56 +104,62 @@ const Countdown = ({ state }: { state?: CountdownState }) => {
...
             Time left
           </div>
-          {state && (isEnded(state) ? (
-            <Row style={{ width: '100%' }}>
-              <div className="cd-number">This auction has ended</div>
-            </Row>
-          ) : (
-            <Row style={{ width: '100%', flexWrap: 'nowrap' }}>
-              {state && state.days > 0 && (
+          {state &&
```
Comment (bhgames) [inline js/packages/web/src/components/AuctionNumbers/index.tsx:107]: Just my formatter formatting things better than your formatter

File: js/packages/web/src/views/auctionCreate/index.tsx [hunk 3]
```diff
@@ -312,6 +332,40 @@ export const AuctionCreateView = () => {
         ? attributes.items[0]
         : attributes.participationNFT,
       QUOTE_MINT,
+      new PriceFloor({
+        type: attributes.priceFloor
+          ? PriceFloorType.Minimum
+          : PriceFloorType.None,
+        hash: new U832ArrayBorshHack({
```
Comment (gching) [inline js/packages/web/src/views/auctionCreate/index.tsx:339]: Should we create a static method on `U832ArrayBorshHack` so we could do something like `U832ArrayBorshHack.fromArray(asArray)` or `U832ArrayBorshHack.create(asArray)`?

File: js/packages/web/src/components/AuctionNumbers/index.tsx (hunk 2 above)
Comment (bartosz-lipinski) [reply js/packages/web/src/components/AuctionNumbers/index.tsx:107]: commit the config so its the same....

File: rust/auction/program/src/processor/place_bid.rs [hunk 4]
```diff
@@ -235,7 +235,7 @@ pub fn place_bid<'r, 'b: 'r>(
             args.amount,
             min[0]
         );
-        if args.amount <= min[0] {
+        if args.amount < min[0] {
```
Comment (bartosz-lipinski) [inline rust/auction/program/src/processor/place_bid.rs:238]: 🤔

File: js/packages/web/src/views/auctionCreate/index.tsx (hunk 3 above)
Comment (bhgames) [reply js/packages/web/src/views/auctionCreate/index.tsx:339]: yeah not a bad call. @bartosz-lipinski is gonna first see if there is a way borsh can just handle this case better so we don't even need this, but if there isn't, I'll add that.

File: rust/auction/program/src/processor/place_bid.rs (hunk 4 above)
Comment (bhgames) [reply rust/auction/program/src/processor/place_bid.rs:238]: If the minimum is 1 and you bid 1, you can't bid it. With this fix, you can now bid 1. Nothing special here. :)

File: js/packages/web/src/components/AuctionNumbers/index.tsx (hunk 2 above)
Comment (bhgames) [reply js/packages/web/src/components/AuctionNumbers/index.tsx:107]: yeah we probably should

File: js/packages/common/src/actions/auction.ts [hunk 5]
```diff
@@ -117,13 +117,24 @@ export enum PriceFloorType {
...
+    type: PriceFloorType;
+    hash?: Uint8Array;
+    minPrice?: BN;
+  }) {
     this.type = args.type;
-    this.hash = args.hash;
+    this.hash = args.hash || new Uint8Array(32);
+    if (this.type === PriceFloorType.Minimum) {
+      if (args.minPrice) {
+        this.hash.set(args.minPrice.toArrayLike(Buffer, 'le', 8), 0);
```
Comment (bhgames) [inline js/packages/common/src/actions/auction.ts:135]: So this sets the first 8 entries, and leaves the rest as 0s?

Comment (bartosz-lipinski) [reply js/packages/common/src/actions/auction.ts:135]: correct.

Comment (bhgames) [reply js/packages/common/src/actions/auction.ts:135]: Love it. Let's merge this bad boy.

**Comment by bhgames** [comment]
@bartosz-lipinski let me know when you've resolved that react issue you said you had

**Pull request merged**